### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by haszi**
This PR fixes existing tests with the following changes:

 - refactor existing `setup.php` to use existing autoloader and initialize basic configuration options
 - add test format from `php` to `xhtml` tests
 - use existing `TestRender` for all tests (as opposed to custom renderer in each one)
 - fix minor format specific issues with dynamic properties and uninitialized arrays
 - update all tests of which the underlying formatting has changed in Phd

### CODE REVIEW

File: phpdotnet/phd/Package/Generic/XHTML.php [hunk 1]
```diff
@@ -832,7 +833,7 @@ public function format_container_chunk_below($open, $name, $attrs, $props) {
 
         $toc = '<ol>';
         $desc = "";
-        if (!in_array($id, $this->TOC_WRITTEN)) {
+        if (!in_array($id, $this->TOC_WRITTEN ?? [])) {
```
Comment (Girgias) [inline phpdotnet/phd/Package/Generic/XHTML.php:836]: I was going to suggest
```suggestion
        if ($this->TOC_WRITTEN && !in_array($id, $this->TOC_WRITTEN)) {
```

But I'm guessing `TOC_WRITTEN` may be undef?

File: tests/php/bug49102-1.phpt [hunk 2]
```diff
@@ -42,73 +46,40 @@ Content:
...
-             ( <span class="methodparam"><span class="type"><a href="language.pseudo-types.html#language.types.mixed" class="type mixed">mixed</a></span> <code class="parameter">$index</code></span>
-         )</div>
-
-        <div class="methodsynopsis dc-description">
-            <span class="type">mixed</span> <span class="methodname"><strong>SplDoublyLinkedList::offsetGet</strong></span>
-             ( <span class="methodparam"><span class="type"><a href="language.pseudo-types.html#language.types.mixed" class="type mixed">mixed</a></span> <code class="parameter">$index</code></span>
-         )</div>
+        <div class="methodsynopsis dc-description"><span class="methodname"><strong>SplDoublyLinkedList::bottom</strong></span>(): <span class="type"><a href="language.types.declarations.html#language.types.declarations.mixed" class="type mixed">mixed</a></span></div>
+
+        <div class="methodsynopsis dc-description"><span class="methodname"><strong>SplDoublyLinkedList::count</strong></span>(): <span class="type">int</span></div>
```
Comment (Girgias) [inline tests/php/bug49102-1.phpt:74]: Showcasing how are links for primitive types are broken :D

File: tests/xhtml/TestChunkedXHTML.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        echo "Filename: " . basename($filename) . "\n";
+        echo "Content:" . $content . "\n";
+    }
+}
+
+/*
+* vim600: sw=4 ts=4 syntax=php et
+* vim<600: sw=4 ts=4
+*/
```
Comment (Girgias) [inline tests/xhtml/TestChunkedXHTML.php:39]: Drop this maybe?

File: phpdotnet/phd/Package/Generic/XHTML.php [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <?php
 namespace phpdotnet\phd;
 
+#[\AllowDynamicProperties]
```
Comment (Girgias) [inline phpdotnet/phd/Package/Generic/XHTML.php:4]: This is sad :(

**Review by Girgias — COMMENTED** [review]
I suppose we could maybe actually run it in CI now :D

**Comment by haszi** [comment]
> I suppose we could maybe actually run it in CI now :D

I actually wanted to ask you about that once this PR was merged. :-)

File: phpdotnet/phd/Package/Generic/XHTML.php (hunk 1 above)
Comment (haszi) [reply phpdotnet/phd/Package/Generic/XHTML.php:836]: There are some dynamic properties in some of these files...

Comment (Girgias) [reply phpdotnet/phd/Package/Generic/XHTML.php:836]: Ah right :|

File: tests/php/bug49102-1.phpt (hunk 2 above)
Comment (haszi) [reply tests/php/bug49102-1.phpt:74]: Oops, they sure are. :-) I'll take look in a follow up PR.

File: tests/xhtml/TestChunkedXHTML.php (hunk 3 above)
Comment (haszi) [reply tests/xhtml/TestChunkedXHTML.php:39]: Just about every files has these lines... Maybe a follow up PR removing all of these at once?

File: tests/php/bug49102-1.phpt (hunk 2 above)
Comment (Girgias) [reply tests/php/bug49102-1.phpt:74]: I had https://github.com/php/phd/pull/33 which I started, *checks notes*, years ago :|

File: phpdotnet/phd/Package/Generic/XHTML.php (hunk 4 above)
Comment (haszi) [reply phpdotnet/phd/Package/Generic/XHTML.php:4]: As I said above, there are some dynamic properties in these files. :-)  I'll make a note of this for a follow up PR.

File: tests/xhtml/TestChunkedXHTML.php (hunk 3 above)
Comment (Girgias) [reply tests/xhtml/TestChunkedXHTML.php:39]: Possibly a good idea yes, well follow-up it is

Comment (haszi) [reply tests/xhtml/TestChunkedXHTML.php:39]: Done.

**Pull request merged**